src/webgpu/api/operation/rendering/draw.spec.ts

```diff
@@ -1,15 +1,27 @@
 export const description = `
 Tests for the general aspects of draw/drawIndexed/drawIndirect/drawIndexedIndirect.
 
-TODO: plan and implement
-- draws (note bind group state is not tested here):
-    - various zero-sized draws
-    - draws with vertexCount not aligned to primitive topology (line-list or triangle-list) (should not error)
-    - index buffer is {unset, set}
```

Comment:
I'm not sure I understood this.  You can't have the index buffer not set for indexed draws. Was this intended to be set/unset for non-indexed draws? I was thinking that ought to go in render/state_tracking.spec.ts

Same for used/not-used vertex buffers. I think most of that would go in render/state_tracking.spec.ts, but we can have something basic here that tests you can have extra stuff bound.

Reply:
Ok I added something to state_tracking.spec.ts